libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapWayName.java

```diff
@@ -79,17 +79,27 @@ boolean removeOnWayNameChangedListener(OnWayNameChangedListener listener) {
   }
 
   void onStart() {
-    if (navigation != null) {
-      navigation.registerRouteProgressObserver(progressChangeListener);
-    }
+    registerObservers();
   }
 
   void onStop() {
     if (isTaskRunning()) {
       filterTask.cancel(true);
     }
+    unregisterObservers();
+  }
+
+  private void registerObservers() {
+    if (navigation != null) {
+      navigation.registerRouteProgressObserver(progressChangeListener);
+      navigation.registerLocationObserver(progressChangeListener);
```

Comment:
need to register the location observer, otherwise it will never pass the check in method `hasValidProgressData()` so the way name view could not show correct info.